### ISSUE TITLE
PipaOS 5.0 "stable" release does not work on RasPi 3B+

### ISSUE THREAD

**Issue by Neex101** [pipaos/pipaos#3]
Doesn't boot.

**Comment by crossmax** [comment]
Yes, I've the same problem. I had a kivypie image on pi3 but don't work on pi3B+. I read that kivypie is based on pipaOS and pipaOS on Raspbian Stretch, and it is assumed that Raspbian Stretch is compatible with pi3B+. But clearly it is not true

**Comment by Neex101** [comment]
So the 'nightly' build of PipaOS-5.0 does boot. I can also get Kivy on there (from Kivy.org), but Kivy crashes when running a program :-( 
Kivy DOES work directly on a Rasbian Stretch Lite build, but for the finished result it takes Rasbian Stretch Lite takes ages to boot and no nice splash screen...

**Comment by crossmax** [comment]
Sounds so bad :disappointed: For your comment it seems that there is no perfect combination to run kivy in pi3B+ and fast boot

**Comment by crossmax** [comment]
I got a kivy application to work on nightly build of PipaOS, but I can't that works (although I've been trying things for days) if I run it at startup. I'm tried with rc.local (enable it before) and as systemd service. Run app on boot, the error returned is "Kivy unable to get a Window". Anyway, if the app don't work on boot, kivy+pipaos is useless for me.

**Comment by skarbat** [comment]
pipaOS 6.0 is now released: http://pipaos.mitako.eu/download/pipaos-stretch-6.0.img.gz

**Comment by Neex101** [comment]
May I be the first to thank @skarbat for your efforts :-) Very much appreciated sir. I'll try immediately. Do you know if Kivy will run on it assuming it's a 3B+? (the previous nightly builds after 5.0 worked on a 3B+, but Kivy would crash)

**Comment by Neex101** [comment]
pipaOS 6.0 release booting fine on my 3B+.
Building kivy now...

**Comment by Neex101** [comment]
Unfortunately Kivy crashes on PipaOS 6.0 release. I have documented my build steps [here](https://github.com/YetiTool/console-raspi3b-plus-platform/blob/master/spikes/kivy_onto_pipaOS6.md). 
Error is `[CRITICAL] [App         ] Unable to get a Window, abort.`
Should I just wait for a KivyPie 1.1 release? Many thanks.

**Comment by Neex101** [comment]
For anyone looking for a temporary solution to getting a 3B+ running with Kivy and 
* Silent boot, 
* splashscreen, 
* autostart,
I've put my build steps for a platform [here](https://github.com/YetiTool/console-raspi3b-plus-platform).
This isn't as good a solution as a fix to KivyPie/PipaOS (slower boot, fewer tools, less of a production solution etc) but does work, and may help some of you in the 3B+ Kivy hole!!

**Comment by pietervandermeer** [comment]
@Neex101 How did you compile Kivy? I recall there's an option to force an SDL backend or so? This way I got Kivy working on the Tinkerboard, for instance..

**Comment by pietervandermeer** [comment]
Maybe this will help ??

1) Recompile SDL2 with KMS/DRM support
Follow instructions for "Linux" on https://wiki.libsdl.org/Installation#Linux.2FUnix

./configure --disable-video-rpi --disable-video-opengl --enable-video-opengles2 --disable-video-mir --enable-video-kmsdrm && make && make install

2) Launch Kivy app with KIVY_GL_BACKEND=sdl2 as an environment variable

3) Profit

**Comment by crossmax** [comment]
Hi.
Any advance with PipaOS on pi3B+ and Kivy? PipaOS was a good solution for headless device in production enverionment
